src/pages/index.js

```diff
@@ -92,38 +124,106 @@ function getCardElement(data) {
     previewModalImageEl.alt = data.name;
   });
 
-  const cardDeleteBtn = cardElement.querySelector(".card__delete-btn");
   cardDeleteBtn.addEventListener("click", () => {
-    cardElement.remove();
+    selectedCard = cardElement;
+    selectedCardId = cardId;
+    openModal(deleteModal);
   });
 
   return cardElement;
 }
 
-// Handle profile edit form submission
+function handleDeleteSubmit() {
+  if (selectedCard && selectedCardId) {
+    const submitButton = deleteBtn;
+    submitButton.textContent = "Deleting...";
+    submitButton.disabled = true;
+
+    api.deleteCard(selectedCardId)
+      .then(() => {
+        selectedCard.remove();
+        closeModal(deleteModal);
+      })
+      .catch(console.error)
+      .finally(() => {
+        submitButton.textContent = "Delete";
+        submitButton.disabled = false;
+        selectedCard = null;
```

Comment:
```js

        selectedCard = null;
        selectedCardId = null;
```

You need to reset the data only in `.then`, to be able to send a request again if there is an error


```js
    api
      .deleteCard(selectedCardId)
      .then(() => {
        selectedCard.remove();
        closeModal(deleteModal);
        selectedCard = null;  // here
        selectedCardId = null;  // here
      })
      .catch(console.error)
      .finally(() => {
        submitButton.textContent = "Delete";
        submitButton.disabled = false;
      });

```